wait/sql.go

```diff
@@ -79,3 +93,75 @@ func (w *waitForSql) WaitUntilReady(ctx context.Context, target StrategyTarget)
 		}
 	}
 }
+
+type waitForSqlWithHost struct {
+	URL            func(port nat.Port, host string) string
+	Driver         string
+	Port           nat.Port
+	startupTimeout time.Duration
+	PollInterval   time.Duration
+}
+
+//Timeout sets the maximum waiting time for the strategy after which it'll give up and return an error
+func (w *waitForSqlWithHost) Timeout(duration time.Duration) *waitForSqlWithHost {
```

Comment:
In PR #322 we are considering to change the name of this method to `WithTimeout` in the other wait strategies.
It could  be renamed here as well

Reply:
I think that's a great change, I'll do that when I get a chance to.